pkg/install/disableupdates_test.go

```diff
@@ -0,0 +1,58 @@
+package install
+
+import (
+	"context"
+	"testing"
+
+	v1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/client-go/config/clientset/versioned/fake"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestDisableUpdates(t *testing.T) {
+	ctx := context.Background()
+
+	versionName := "version"
+
+	configcli := fake.NewSimpleClientset()
+	configcli.ConfigV1().ClusterVersions().Create(
+		&v1.ClusterVersion{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: versionName,
+			},
+		},
+	)
+
+	cv := getClusterVersion(t, configcli, versionName)
+	cv.Spec.Upstream = "RemoveMe"
+	cv.Spec.Channel = "RemoveMe"
+
+	_, err := configcli.ConfigV1().ClusterVersions().Update(cv)
+	if err != nil {
+		t.Error(err)
+	}
+
+	i := &Installer{
+		configcli: configcli,
+	}
+	err = i.disableUpdates(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+
+	cv = getClusterVersion(t, i.configcli.(*fake.Clientset), versionName)
+	if cv.Spec.Upstream != "" {
+		t.Errorf("cv.Spec.Upstream was not cleared. Got %s, want empty string", cv.Spec.Upstream)
+	}
+	if cv.Spec.Channel != "" {
+		t.Errorf("cv.Spec.Channel was not cleared. Got %s, want empty string", cv.Spec.Channel)
+	}
+}
+
+func getClusterVersion(t *testing.T, configcli *fake.Clientset, versionName string) *v1.ClusterVersion {
```

Comment:
I think you can use `configclient.Interface` here instead of `*fake.Clientset`: both fake and client should implement the inferface.

This way you will not need to convert the type on line 43.

Reply:
@m1kola Changed